### PR TITLE
ci: automate discovery types 

### PR DESCRIPTION
**Pull request by alvarowolfx**
Towards internal b/309836530 and supersedes #1366

### CODE REVIEW

File: .github/workflows/discovery.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  workflow_dispatch:    
+name: Update Discovery Generated Types
+jobs:
+  sync:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # v4.1.6
+      - uses: actions/setup-node@60edb5dd545a775178f52524783378180af0d1f8 # v4.0.2
+        with:
+          node-version: 16
```
Comment (leahecole) [inline .github/workflows/discovery.yaml:14]: will these be managed by renovate or dependabot?

File: package.json [hunk 2]
```diff
@@ -55,7 +56,9 @@
     "arrify": "^2.0.1",
     "big.js": "^6.0.0",
     "duplexify": "^4.0.0",
+    "execa": "^5.0.0",
     "extend": "^3.0.2",
+    "gaxios": "^6.0.3",
```
Comment (leahecole) [inline package.json:61]: should these be dev dependencies instead of regular ones?

File: scripts/submit-discovery-pr.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  console.log(foundChanges.join('\n'));
+
+  if (foundChanges.length === 0) {
+    console.log('No changes found');
+    return;
+  }
+
+  if (process.env.GITHUB_ACTIONS) {
+    await execa('git', ['config', 'user.email', 'yoshi-automation@google.com']);
+    await execa('git', ['config', 'user.name', 'Yoshi Automation']);
```
Comment (leahecole) [inline scripts/submit-discovery-pr.js:43]: @sofisl is this the most up to date git username and email we use for this kind of stuff?

File: .github/workflows/discovery.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+on:
+  schedule:
+    - cron: '0 12 * * TUE'
+  pull_request: # REMOVE BEFORE MERGING
```
Comment (leahecole) [inline .github/workflows/discovery.yaml:4]:
```suggestion
  pull_request: # REMOVE BEFORE MERGING
```

Can this be removed?

File: .github/workflows/discovery.yaml (hunk 1 above)
Comment (alvarowolfx) [reply .github/workflows/discovery.yaml:14]: yeah, renovate/dependabot managed those dependencies. See example here https://github.com/googleapis/google-api-nodejs-client/pull/3485

File: package.json (hunk 2 above)
Comment (alvarowolfx) [reply package.json:61]: ah good catch, I forgot the `--dev` flag when I added then

File: .github/workflows/discovery.yaml (hunk 4 above)
Comment (alvarowolfx) [reply .github/workflows/discovery.yaml:4]: right now I only added it so we can trigger it here on this PR and to make sure it works. But will remove before merging

File: scripts/submit-discovery-pr.js (hunk 3 above)
Comment (alvarowolfx) [reply scripts/submit-discovery-pr.js:43]: for context, I got from https://github.com/googleapis/google-api-nodejs-client/blob/main/src/generator/synth.ts

File: scripts/submit-discovery-pr.js [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+  if (!githubToken) {
+    throw new Error('please include a GITHUB_TOKEN');
+  }
+
+  try {
+    // Open the pull request
+    await gaxios.request({
+      method: 'POST',
+      headers: {
+        Authorization: `token ${githubToken}`,
```
Comment (chingor13) [inline scripts/submit-discovery-pr.js:64]: This looks like it should be a bearer token
```suggestion
        Authorization: `Bearer ${githubToken}`,
```

**Review by sofisl — APPROVED** [review]

File: scripts/submit-discovery-pr.js (hunk 3 above)
Comment (sofisl) [reply scripts/submit-discovery-pr.js]: yes!

**Pull request merged**